vllm/model_executor/models/gemma3_mm.py

```diff
@@ -749,13 +754,22 @@ def prepare_attn_masks(
 
             global_attn_masks.append(global_attn_mask)
             if self.sliding_window is not None:
-                # Create a local causal mask with sliding window (1024).
-                local_attn_mask = torch.ones_like(global_attn_mask)
-                local_attn_mask = torch.tril(local_attn_mask,
-                                             diagonal=-self.sliding_window)
-                local_attn_mask = torch.where(local_attn_mask == 0,
-                                              global_attn_mask, float("-inf"))
-                local_attn_masks.append(local_attn_mask)
+                if is_hpu and self.use_fsdpa_window and (
+                        self.fsdpa_window_slice_size
+                        != 0) and (seq_len % self.fsdpa_window_slice_size
+                                   == 0):
+                    # In HPU, no need to create local attn_mask
+                    # if FusedSDPA with window_size kernel is supported.
+                    local_attn_masks = None
+                else:
+                    # Create a local causal mask with sliding window (1024).
+                    local_attn_mask = torch.ones_like(global_attn_mask)
+                    local_attn_mask = torch.tril(local_attn_mask,
+                                                 diagonal=-self.sliding_window)
+                    local_attn_mask = torch.where(local_attn_mask == 0,
+                                                  global_attn_mask,
+                                                  float("-inf"))
+                    local_attn_masks.append(local_attn_mask)
```

Comment:
suggest to not change original gemma3_mm.py modeling codes to keep it identical with vllm upstream version.
To decide if use local_attn_masks, please handle that in hpu_attn.py